models/clm/readme.rst

```diff
@@ -1,40 +1,195 @@
-requirements out of the model code, there are a few SourceMods needed to run DART from within CESM. Appropriate
-SourceMods for each CESM version are available at http://www.image.ucar.edu/pub/DART/CESM and should be unpacked into
-your HOME directory. They will create a ``~/cesm_?_?_?`` directory with the appropriate SourceMods structure. The
-ensuing scripts require these SourceMods and expect them to be in your HOME directory.
-
-Our notes on how to set up, configure, build, and run CESM for an assimilation experiment evolved into scripts. These
-scripts are not intended to be a 'black box'; you will have to read and understand them and modify them to your own
-purpose. They are heavily commented -- in keeping with their origins as a set of notes. If you would like to offer
-suggestions on how to improve those notes - please send them to dart@ucar.edu - we'd love to hear them.
+     dimension_name = get_dim_name(dom_id, var_id, jdim)
+     select case ( trim(dimension_name) )
+            case ("gridcell","lon","lat")
+               related = .true.
+            case ("lndgrid")
+               related = .true.
+            case ("landunit")
+               if ( land1d_ityplun(indices(jdim)) == ilun_vegetated_or_bare_soil ) related = .true.
+               if ( land1d_ityplun(indices(jdim)) == ilun_crop                   ) related = .true.
+            case ("column")
+               if ( cols1d_ityplun(indices(jdim)) == icol_vegetated_or_bare_soil ) related = .true.
+               if ( cols1d_ityplun(indices(jdim)) == icol_crop                   ) related = .true.
+            case ("pft")
+               related = .true.
+            case default
+     end select
+  
+     ! Since variables can use only one of these dimensions,
+     ! there is no need to check the other dimensions. 
+     if (related) exit RELATEDLOOP
+  
+  enddo RELATEDLOOP
+
+
+Snow Data Assimilation
+~~~~~~~~~~~~~~~~~~~~~~
+
+The *prognostic* variables for snow (i.e. the ones that impact the forecast) 
+are the ones that have layers. The snow observations are typically without 
+explicit depths and are essentially column-integrated quantities like snow 
+water equivalent (SWE - CLM variable *H2OSNO*) or snow depth 
+(CLM variable *SNOW_DEPTH*).  These CLM *diagnostic* variables  
+simplify the forward operator and have been part of the DART state. 
+However, updating the prognostic variables 
+(*T_SOISNO, H2OSOI_LIQ, H2OSOI_ICE, DZSNO, ZSNO, ZISNO*)
+through their ensemble covariance with the update to *H2OSNO*, for example
+will generally not result in a posterior SWE (calculated from the prognostic 
+variables) that matches the posterior SWE in *H2OSNO*. Consequently, a 
+partitition function should be used to redistribute the posterior SWE into
+the appropriate prognostic variables. **This is under development.**
+Note that we have not attempted to include any of the snow property
+variables (grain radius, carbon content, etc) in the DART state.
+
```

Comment:
I think what we mean is that we 'have not attempted to include any *non-water related* snow property variables (grain radius, carbon content, etc) in the DART state.

Reply:
grain radius could be construed as a water-related property - so I am leaving this.